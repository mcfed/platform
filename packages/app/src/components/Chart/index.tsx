import React, {useLayoutEffect, useRef} from 'react';
import * as G2Plot from '@antv/g2plot';
import DrawScene from '../Chart/map';
import {Editor, defaultConfigs} from '@antv/g2plot-schemas';
import {isEqual} from '@antv/util';
interface PG2PloatChat {
  id: string;
  data: any;
  type?: string;
  configs: any;
}

/**
 * 去除掉所有默认的配置
 * @param configs - 配置
 * @param defaultCfgs - 默认配置
 */
function shake(configs: any, defaultCfgs: any) {
  const result: any = {};
  if (!defaultCfgs) return configs;
  for (const [key, value] of Object.entries(configs)) {
    if (typeof value === 'object' && !Array.isArray(value) && value !== null) {
      const newValue = shake(value, defaultCfgs[key]);
      if (Object.keys(newValue).length !== 0) result[key] = newValue;
    } else if (Array.isArray(value)) {
      if (!isEqual(configs[key], defaultCfgs[key])) {
        result[key] = value;
      }
    } else {
      if (configs[key] !== defaultCfgs[key]) {
        result[key] = value;
      }
    }
  }
  return result;
}

export const RG2PlotChart = React.forwardRef((props: PG2PloatChat) => {
  const eleRef = useRef<HTMLDivElement>(null);
  const chartType = props.configs.type || props.type;
  //@ts-ignore
  const {xAxis, yAxis, height, width, label, ...config} = shake(
    props.configs,
    defaultConfigs[chartType]
  );
  console.log(config);
  useLayoutEffect(function() {
    //@ts-ignore
    if (chartType == 'map') {
      const mapScene = new DrawScene(eleRef.current, props.data, config);
      mapScene.draw();
    } else {
      //@ts-ignore
      new G2Plot[chartType](eleRef.current, {
        ...config,
        data: props.data
      }).render();
    }
  });

  return <div ref={eleRef}></div>;
});
