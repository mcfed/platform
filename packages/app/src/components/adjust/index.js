import {registerAdjust} from '@antv/adjust';
import Scatter from './scatter';

registerAdjust('Scatter', Scatter);
