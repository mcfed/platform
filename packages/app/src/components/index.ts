export enum MenuKeyEnum {
  profile = 'profile',
  passwd = 'passwd',
  logout = 'logout'
}
