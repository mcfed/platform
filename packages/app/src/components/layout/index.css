.flex-layout {
  display: flex;
  flex: 1;
}

.flex-pane {
  flex: 1;
  display: flex;
  flex-direction: row;
  overflow: hidden;
}
