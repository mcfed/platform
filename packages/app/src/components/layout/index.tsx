import React, {CSSProperties, ReactNode} from 'react';
import './index.css';

export interface PLayout {
  className?: string;
  direction?: 'column' | 'row';
  children: ReactNode;
  style?: CSSProperties;
}
const DLayout: PLayout = {
  className: '',
  direction: 'row',
  children: null
};

export function RFLayout(props: PLayout = DLayout) {
  return (
    <div
      className={['flex-layout', props.className].join(' ')}
      style={{...props.style, flexDirection: props.direction}}>
      {props.children}
    </div>
  );
}

export interface PPane {
  className?: string;
  children: ReactNode;
  collaplse?: Boolean;
  style?: CSSProperties;
  direction?: 'left' | 'right' | 'top' | 'bottom';
}

export function RFPane(props: PPane) {
  function renderCollapse(props: PPane) {
    if (props.collaplse) {
      return <div className={['flex-layout', 'collapse'].join('-')}>&gt;</div>;
    }
  }
  return (
    <div
      className={['flex-pane', props.className].join(' ')}
      style={props.style}>
      {renderCollapse(props)}
      {props.children}
    </div>
  );
}
