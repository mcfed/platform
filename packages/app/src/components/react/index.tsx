import {HtmlComponent, IList, ListItem} from '@antv/component';
import React from 'react';
import ReactDOM from 'react-dom';
import {List, Typography} from 'antd';

export class ReactG2Component extends HtmlComponent {
  renderReact() {
    const items = this.get('items');
    return (
      <List
        bordered
        dataSource={items}
        renderItem={item => (
          <List.Item>
            <Typography.Text mark>[ITEM]</Typography.Text> {item}
          </List.Item>
        )}
      />
    );
  }
  render() {
    const parent: HTMLElement = this.get('parent');
    const curContainer: HTMLElement = this.get('container');
    this.set('container', curContainer);
    ReactDOM.render(this.renderReact(), curContainer);
  }
}
