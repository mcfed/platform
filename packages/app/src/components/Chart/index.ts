import {RG2PlotChart} from './G2PloatChart';
import {registerComponentController} from '@antv/g2';
import Map from './map';
import '../adjust';
import '../connector/fetch';

// registerComponentController('Map',Map)
export {RG2PlotChart};
