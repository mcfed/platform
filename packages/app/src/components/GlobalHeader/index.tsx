import React, {ReactNode, ComponentClass, Fragment} from 'react';
import { Menu, Dropdown, Avatar} from 'antd';

import {ClickParam} from 'antd/lib/menu';
import {BasicProps} from 'antd/lib/layout/layout';
import './style.css';

export interface GlobalHeaderProps extends BasicProps {
  avator?: ReactNode;
  logo?: ReactNode;
  user?: any;
  onMenuSelect?(key: MenuKeyEnum): void;
}

export enum MenuKeyEnum {
  profile = 'profile',
  passwd = 'passwd',
  logout = 'logout'
}


export class GlobalHeader extends React.Component<GlobalHeaderProps> {
  getMenu() {
    const {user} = this.props;
    const disabled = user.userOrigin !== 'MC_CENTER';
    return (
      <Menu onClick={this.handleMenuClick.bind(this)}>
        <Menu.Item key={MenuKeyEnum.profile}>基础资料</Menu.Item>
        <Menu.Item key={MenuKeyEnum.passwd} disabled={disabled}>
          修改密码
        </Menu.Item>
        <Menu.Divider />
        <Menu.Item key={MenuKeyEnum.logout}>退出登录</Menu.Item>
      </Menu>
    );
  }

  handleMenuClick({key}: ClickParam) {
    const {onMenuSelect} = this.props;
    if (typeof onMenuSelect === 'function') onMenuSelect(key as MenuKeyEnum);
  }

  getLogo() {
    const {logo} = this.props;
    if (!logo) return null;
    if (typeof logo === 'string') return <img alt='logo' src={logo} />;
    return logo;
  }

  render() {
    const {avator, children, className, ...otherProps} = this.props;
    const {logo} = otherProps;
    return (
      <Fragment>
        {logo && <div className='header-logo'>{this.getLogo()}</div>}
        <div className='header-content'>{children}</div>
        <Dropdown
          className='header-icon'
          overlay={this.getMenu()}
          placement='bottomLeft'>
          <Avatar>{avator}</Avatar>
        </Dropdown>
      </Fragment>
    );
  }
}

(GlobalHeader as ComponentClass).defaultProps = {
  avator: 'User'
};
