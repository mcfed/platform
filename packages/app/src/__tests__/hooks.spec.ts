import {renderHook} from '@testing-library/react-hooks';
import {Dispatch} from 'redux';
import {MenuKeyEnum} from '../components';

import {
  useHandleMenuSelect,
  useLogoutAction
} from '../layouts/hooks';

describe('layout hooks', () => {
  const action = {type: 'test'};
  const history: any = {push: jest.fn()};
  const dispatch: Dispatch = jest.fn();

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('useLogoutAction', () => {
    const {result} = renderHook(() => useLogoutAction());
    expect(result).toMatchSnapshot();
  });

  // it.skip('useDidMount|无token', () => {
  //   renderHook(() => {
  //     useDidMount({appReducer: {}});
  //   });
  //   expect(dispatch).toHaveBeenCalledWith(action);
  // });

  // it.skip('useDidMount|有token', () => {
  //   renderHook(() => {
  //     useDidMount({appReducer: {token: 'test'}});
  //   });
  //   expect(dispatch).toHaveBeenCalledTimes(0);
  // });

  it('useHandleMenuSelect|profile', () => {
    const {result} = renderHook(() =>
      useHandleMenuSelect({userId: 123, history, action, dispatch})
    );
    result.current(MenuKeyEnum.profile);
    expect(history.push).toHaveBeenCalledWith('/my/profile');
  });

  it('useHandleMenuSelect|passwd', () => {
    const {result} = renderHook(() =>
      useHandleMenuSelect({userId: 123, history, action, dispatch})
    );
    result.current(MenuKeyEnum.passwd);
    expect(history.push).toHaveBeenCalledWith('/my/profile/config/123/edit');
  });

  it('useHandleMenuSelect|logout', () => {
    const {result} = renderHook(() =>
      useHandleMenuSelect({userId: 123, history, action, dispatch})
    );
    result.current(MenuKeyEnum.logout);
    expect(dispatch).toHaveBeenCalledWith(action);
  });
});
