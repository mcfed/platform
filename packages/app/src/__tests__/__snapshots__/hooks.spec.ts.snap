// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout hooks useLogoutAction 1`] = `
Object {
  "current": Object {
    "payload": Object {},
    "type": "@@MIDDLEWARE/fetchLogouting",
  },
  "error": undefined,
}
`;
